app/cdap/components/Connections/Browser/GenericBrowser/index.tsx

```diff
@@ -0,0 +1,156 @@
+      marginTop: '15px',
+    },
+    loadingContainer: {
+      display: 'flex',
+      justifyContent: 'center',
+      alignItems: 'center',
+      height: '100%',
+    },
+  };
+});
+
+export function GenericBrowser({ selectedConnection }) {
+  const loc = useLocation();
+  const queryParams = new URLSearchParams(loc.search);
+  const pathFromUrl = queryParams.get('path') || '/';
+  const [entities, setEntities] = React.useState([]);
+  const [loading, setLoading] = React.useState(true);
+  const [error, setError] = React.useState(null);
+  const [path, setPath] = React.useState(pathFromUrl);
+  const classes = useStyle();
+
+  const fetchEntities = async () => {
+    try {
+      const res = await exploreConnection({
+        connectionid: selectedConnection,
+        path,
+      });
+      setEntities(res.entities);
+      setLoading(false);
+    } catch (e) {
+      setLoading(false);
+      setError(`Failed to explore connection : "${e.message}"`);
+    }
+  };
+  const onExplore = (entityName) => {
+    if (path === '/') {
+      setPath(`/${entityName}`);
+    } else {
+      setPath(`${path}/${entityName}`);
+    }
+    setLoading(true);
+  };
+  React.useEffect(() => {
+    if (isNilOrEmptyString(selectedConnection)) {
+      return setLoading(false);
+    }
+    fetchEntities();
+  }, [selectedConnection, path]);
+
+  React.useEffect(() => {
+    const query = new URLSearchParams(loc.search);
+    const urlPath = query.get('path') || '/';
+    if (path !== urlPath) {
+      setPath(urlPath);
+    }
+  }, [loc]);
+  if (loading) {
+    return (
+      <div className={classes.loadingContainer}>
+        <LoadingSVG />
+      </div>
+    );
+  }
+
+  if (error) {
+    throw error;
+  }
+  if (!Array.isArray(entities) || (Array.isArray(entities) && !entities.length)) {
+    return <div>No entities available</div>;
+  }
+  let headers = ['Name', 'Type'];
+  const properties = objectQuery(entities, 0, 'properties');
+  headers = [...headers, ...properties.map((prop) => capitalize(prop.key))];
+  const columnTemplate = headers.map(() => '1fr').join(' ');
+  const getPath = (suffix) => (path === '/' ? `/${suffix}` : `${path}/${suffix}`);
+  return (
+    <Table columnTemplate={columnTemplate} classes={{ grid: classes.grid }}>
+      <TableHeader>
+        <TableRow>
+          {headers.map((header) => {
+            return <TableCell key={header}>{header}</TableCell>;
+          })}
+        </TableRow>
+      </TableHeader>
+      <TableBody>
+        {entities.map((entity, i) => {
+          return (
+            <TableRow
+              key={i}
+              to={`/ns/${getCurrentNamespace()}/connections/${selectedConnection}?path=${getPath(
+                entity.name
+              )}`}
+              onClick={() => onExplore(entity.name)}
+            >
+              <TableCell>
+                <div className={classes.nameWrapper}>
+                  <If condition={entity.type === 'directory'}>
+                    <FolderIcon />
```

Comment:
Nit: we need to a way to determine if other icons should be shown, like a database icon.

Reply:
You are correct. This has to be dynamic for each connection. Right now I am not addressing the icons. Just the base structure. We need to think about icons for categories, connectors, connections and individual entities in each connection. 

The good part is the API already has the information we need for showing icons under a connection while exploring. We will need to address this when we deal with icons as a whole for connections/wrangler interaction.